examples/xt.lua

```diff
@@ -0,0 +1,51 @@
+local xt = require("xtable")
+local action = xt.action
+local family = xt.family
+local match_packet
+local check_packet
+local destroy_packet
+
+local udp = 0x11
+local dns = 0x35
+
+function match_packet(skb)
+	print("match_packet called\n")
+	-- ip header
+	local ipb = skb:getbyte(0)
+	local ihlen = (( ipb << 4 ) & 0xFF ) >> 4
+	local iphdrlen = ihlen * 4
+	local proto = skb:getbyte(9)
+
+	if proto == udp then
+		local srcport = skb:getbyte(iphdrlen) << 8 | skb:getbyte(iphdrlen + 1)
+		local dstport = skb:getbyte(iphdrlen + 2) << 8 | skb:getbyte(iphdrlen + 3)
+		if srcport == dns or dstport == dns then
+			print("DNS packet\n")
+			return action.ACCEPT
+		end
+	end
+
+	return action.DROP
+end
+
+function check_packet()
+	print("check_packet called\n")
+	return 0
```

Comment:
is this return really necessary? I would simplify your example and do something like..

```
local function nop() end

...

local dnsdrop = {
	name= "dnsdrop",
	...
	checkentry = nop,
	destroy = nop,
	...
}
```

Reply:
check packet will be needed for now, destroy can be a nop

Reply:
why? aren't you doing `optint(..., 0)`? it will evaluate to 0 anyway if return none, doesn't it?

Reply:
Ah, yes